src/pages/TeamManagment/UserForm/useUserForm.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { toast } from 'react-toastify';
+
+import { RootState } from '@/store/store';
+import axiosInstance from '@/utils/axiosInstance';
+import { getFirstName, getSecondName } from '@/utils/nameUtils';
+
+import { DecodedToken, UserFormInputs, UserFormProps } from '../types';
+
+export const useUserForm = ({
+  mode,
+  fetchUsers,
+  userId,
+  userData,
+  addUserToList,
+}: UserFormProps) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [role, setRole] = useState(userData?.role || '');
+  const { t } = useTranslation();
+  const authToken = useSelector((store: RootState) => store.auth.token);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    reset,
+    formState: { errors },
+  } = useForm<UserFormInputs>({
+    defaultValues:
+      mode === 'update'
+        ? {
+            fullName: userData?.fullName || '',
+            email: userData?.email || '',
+            phoneNumber: userData?.phoneNumber || '',
+          }
+        : {
+            fullName: '',
+            email: '',
+            phoneNumber: '',
+            role: '',
+          },
+  });
+
+  const closeModal = (): void => {
+    setIsModalOpen(false);
+    reset();
+  };
+
+  const decodeToken = (token: string): DecodedToken | null => {
+    try {
+      const decodedToken = JSON.parse(atob(token.split('.')[1]));
+      return decodedToken;
+    } catch (error) {
+      console.error('Error decoding token:', error);
+      return null;
+    }
+  };
+
+  const sendData = async (formData: UserFormInputs) => {
+    try {
+      const decodedToken = authToken ? decodeToken(authToken) : null;
+
+      if (!decodedToken) {
+        console.error('Invalid token');
+        toast(t('settings.message.invalidToken'), { type: 'error' });
+        return;
+      }
+
+      const companyId = decodedToken.company_id.id;
```

Comment:
why you take company id from token? if user belong to company and you need access to this value from some places, save this info during login in redux

Reply:
fixed it